File: app/src/main/java/com/moa/rxdemo/db/TestData.java
package com.moa.rxdemo.db;

import com.moa.baselib.utils.DateFormatting;
import com.moa.baselib.utils.Randoms;
import com.moa.rxdemo.db.entity.Book;
import com.moa.rxdemo.db.entity.Student;
import com.moa.rxdemo.db.entity.User;

import java.util.ArrayList;
import java.util.List;

public class TestData {


    public static User getUser() {

        User user = new User();
        user.address = "北京市日本镇日本村${randomStr}号";
        user.id = Randoms.randomInt() + "";
        user.name = "小白${randomStr}号";
        user.phone = "1500000000$randomStr";
        return user;
    }

    public static Book getBook(String uid) {

       Book book = new Book();
       book.bookid = Randoms.randomInt() + "";
       book.uid = uid;
       book.bookname = "钓鱼岛是中国的$randomStr";
       book.date = DateFormatting.formatDate(System.currentTimeMillis());
       book.author = "小黑$randomStr";

        return book;
    }

    public static List<Book> getBooks(String uid) {

        List<Book> items = new ArrayList<>();
        int size = (int) (Math.random()*10 + 10);
        for (int i =0;i < size;i ++){
            items.add(getBook(uid));
        }

        return items;
    }


    public static Student getStudent(String uid)

    {
        String randomStr = Randoms.randomInt()+"";
        Student student = new Student();
        student.bookid = randomStr;
        student.uid = uid;
        student.bookname = "钓鱼岛是中国的$randomStr";
        student.date = DateFormatting.formatFull(System.currentTimeMillis());
        student.author = "小黑$randomStr";
        student.desc = "台湾岛也是中国的$randomStr";

        return student;
    }
}

File: app/src/main/java/com/moa/rxdemo/base/db/TestData.kt
package com.moa.rxdemo.base.db

import com.moa.rxdemo.base.db.entity.Book
import com.moa.rxdemo.base.db.entity.Student
import com.moa.rxdemo.base.db.entity.User
import com.moa.rxdemo.utils.DateFormatting
import com.moa.rxdemo.utils.Randoms
import java.util.*

/**
 * 封装测试数据
 *
 * Created by：wangjian on 2018/12/19 09:51
 */
object TestData {

    fun getUser(): User {
        val randomStr = Randoms.randomInt().toString();

        val user = User();
        user.address = "北京市日本镇日本村${randomStr}号"
        user.id = randomStr
        user.name = "小白${randomStr}号"
        user.phone = "1500000000$randomStr"

        return user;
    }

    fun getBook(uid: String): Book {
        val randomStr = Randoms.randomInt().toString();
        val book = Book();
        book.bookid = randomStr;
        book.uid = uid;
        book.bookname = "钓鱼岛是中国的$randomStr";
        book.date = DateFormatting.formatDate(Date().time);
        book.author = "小黑$randomStr";
        book.desc = "台湾岛也是中国的$randomStr";

        return book;
    }

    fun getBooks(uid: String): List<Book>{
        val items = mutableListOf<Book>()
        val size = Math.random()*10 + 10;
        for (item in 0..size.toInt()) {
            items.add(getBook(uid))
        }

        return items;
    }

    fun getStudent(uid: String): Student {
        val randomStr = Randoms.randomInt().toString();
        val student = Student();
        student.bookid = randomStr;
        student.uid = uid;
        student.bookname = "钓鱼岛是中国的$randomStr";
        student.date = DateFormatting.formatDate(Date().time);
        student.author = "小黑$randomStr";
        student.desc = "台湾岛也是中国的$randomStr";

        return student;
    }
}
